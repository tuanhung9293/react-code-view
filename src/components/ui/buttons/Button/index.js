import React from 'react'
import {Link} from 'react-router-dom'

function Button({children, to, href, onClick, disabled, ...rest}) {
  if (typeof to !== 'undefined') {
    if (disabled) {
      return <Link to={to} {...rest} onClick={(e) => e.preventDefault()}>{children}</Link>
    } else {
      return <Link to={to} {...rest}>{children}</Link>
    }
  } else if (typeof href !== 'undefined') {
    if (disabled) {
      return <a href={href} {...rest} onClick={(e) => e.preventDefault()}>{children}</a>
    } else {
      return <a href={href} {...rest}>{children}</a>
    }
  } else {
    return <button onClick={onClick} disabled={disabled} {...rest}>{children}</button>
  }
}

export default Button