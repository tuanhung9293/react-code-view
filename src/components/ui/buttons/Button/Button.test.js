import React from 'react'
import {shallow} from 'enzyme'
import Button from './'

it('renders children', () => {
  expect(shallow(
    <Button>I am your child</Button>
  )).toMatchSnapshot()
})

it('renders as a link when to prop is provided', () => {
  expect(shallow(
    <Button to="/my-path">I am your child</Button>
  )).toMatchSnapshot()
})

it('renders as an anchor when href prop is provided', () => {
  expect(shallow(
    <Button href="https://someurl.com">I am your child</Button>
  )).toMatchSnapshot()
})

it('renders as disabled', () => {
  expect(shallow(
    <Button disabled>I am your child</Button>
  )).toMatchSnapshot()
})

it('renders as disabled also as a link', () => {
  expect(shallow(
    <Button to="/hello" disabled>I am your child</Button>
  )).toMatchSnapshot()
})

it('prevents default when disabled link is clicked', () => {
  const preventDefault = jest.fn()
  const wrapper = shallow(
    <Button to="/hello" disabled>I am your child</Button>
  )
  wrapper.simulate('click', {preventDefault})
  expect(preventDefault).toBeCalled()
})