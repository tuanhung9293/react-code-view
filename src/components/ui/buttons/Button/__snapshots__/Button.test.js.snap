// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as a link when to prop is provided 1`] = `
<Link
  replace={false}
  to="/my-path"
>
  I am your child
</Link>
`;

exports[`renders as an anchor when href prop is provided 1`] = `
<a
  href="https://someurl.com"
>
  I am your child
</a>
`;

exports[`renders as disabled 1`] = `
<button
  disabled={true}
>
  I am your child
</button>
`;

exports[`renders as disabled also as a link 1`] = `
<Link
  onClick={[Function]}
  replace={false}
  to="/hello"
>
  I am your child
</Link>
`;

exports[`renders children 1`] = `
<button>
  I am your child
</button>
`;
